app/assets/javascripts/admin/components/admin-report-table.js.es6

```diff
@@ -79,8 +79,8 @@ export default Ember.Component.extend({
     if (sortLabel) {
       const compare = (label, direction) => {
         return (a, b) => {
-          let aValue = label.compute(a).value;
-          let bValue = label.compute(b).value;
+          let aValue = label.compute(a, { useSortProperty: true }).value;
```

Comment:
We wish to allow Admins to sort the uploads table by file size, yet, our column value is based on the `filesize_label` property which is of type String. We also return the `filesize` property which is a number.

If [you look at this line](https://github.com/discourse/discourse/pull/6809/files#diff-3f97cbb8726f3310e0b0c386dbe89e22R1451), you can see that we are setting the `sort_property` to be `filesize`. Therefore, I found that within the `compute` method, we still end up using the `main property for the column`, not the `sort_property`. 

I exposed this new option to allow depending on the `sort_property` if supplied by the backend.

Reply:
👍